File: app/src/main/java/com/personal/accountantAssistant/utils/ActivityUtils.java
package com.personal.accountantAssistant.utils;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;

import com.personal.accountantAssistant.ui.payments.PaymentsDetailsActivity;
import com.personal.accountantAssistant.ui.payments.entities.Payments;
import com.personal.accountantAssistant.ui.payments.enums.PaymentsType;

import java.util.Objects;

public class ActivityUtils {

    public static void startActivity(final Context packageContext,
                                     final Class<?> activityClass) {
        final Intent intent = new Intent(packageContext, activityClass);
        packageContext.startActivity(intent);
    }

    public static void startPaymentDetailsActivity(final Context packageContext,
                                                   final PaymentsType paymentsType) {
        final Activity activity = parse(packageContext);
        final Payments payment = new Payments();
        payment.setType(paymentsType);
        final Intent activityIntent = new Intent(packageContext, PaymentsDetailsActivity.class);
        activityIntent.putExtra(Constants.ENTITY, payment);
        activity.startActivityForResult(activityIntent, Constants.DETAIL_REQUEST_CODE);
    }

    public static void refreshBy(final Context context) {
        final Activity activity = parse(context);
        Objects.requireNonNull(activity).recreate();
    }

    public static Activity parse(final Context context) {
        return (Activity) context;
    }

    public static Context parse(final Activity activity) {
        return activity.getApplicationContext();
    }
}

File: app/src/main/java/com/personal/accountantAssistant/utils/ActivityUtils.kt
package com.personal.accountantAssistant.utils

import android.app.Activity
import android.content.Context
import android.content.Intent
import com.personal.accountantAssistant.ui.MainActivity

object ActivityUtils {

    @JvmStatic
    fun startMainActivity(packageContext: Context) {
        Intent(packageContext, MainActivity::class.java).apply {
            flags = Intent.FLAG_ACTIVITY_NEW_TASK
            packageContext.startActivity(this)
        }
    }

    @JvmStatic
    fun startActivity(
        packageContext: Context,
        activityClass: Class<*>?
    ) {
        Intent(packageContext, activityClass).apply {
            packageContext.startActivity(this)
        }
    }

    @JvmStatic
    fun parse(context: Context): Activity {
        return context as Activity
    }

    @JvmStatic
    fun parse(activity: Activity?): Context? {
        return activity?.applicationContext
    }
}